src/QueryBuilder/QueryViewer/QueryViewer.js

```diff
@@ -0,0 +1,124 @@
+import React, { memo, useEffect, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { Col, Row, Accordion, AccordionSet, MultiColumnList, Headline } from '@folio/stripes/components';
+import { PrevNextPagination, usePagination } from '@folio/stripes-acq-components';
+import QueryLoader from '../QueryLoader/QueryLoader';
+import { getTableMetadata } from '../helpers';
+
+const QueryViewer = ({
+  showPagination = true,
+  contentData = [],
+  limit = 100,
+  offset = 0,
+  onPageChange,
+  entityType,
+  headline,
+  headlineEnd,
+  visibleColumns,
+  onSetDefaultVisibleColumns,
+  onSetDefaultColumns,
+  accordionHeadline,
+  isInProgress,
+  inProgressTitle,
+  height,
+}) => {
+  const { changePage, pagination } = usePagination({ limit, offset });
+  const {
+    columnMapping,
+    defaultColumns,
+    defaultVisibleColumns,
+  } = useMemo(() => getTableMetadata(entityType), [entityType]);
+
+  useEffect(() => {
+    onPageChange?.(pagination);
+  }, [pagination]);
+
+  // set visible by default columns once
+  useEffect(() => {
+    if (defaultVisibleColumns && defaultColumns) {
+      onSetDefaultColumns?.(defaultColumns);
+      onSetDefaultVisibleColumns?.(defaultVisibleColumns);
+    }
+  }, [defaultVisibleColumns, defaultColumns]);
+
+  const renderTable = () => (
+    <>
+      <Row between="xs">
+        <Col xs={6}>
+          <Headline size="large" margin="none" tag="h3">
+            {headline}
+          </Headline>
+        </Col>
+        {headlineEnd}
+      </Row>
+      <Row>
+        <Col xs={12}>
+          <MultiColumnList
+            contentData={contentData}
+            columnMapping={columnMapping}
+            visibleColumns={visibleColumns}
+            pagingType={null}
+            onNeedMoreData={changePage}
+            height={height}
+          />
+          {showPagination && (
+            <PrevNextPagination
+              {...pagination}
+              totalCount={contentData?.length}
+              disabled={contentData?.length <= limit}
```

Comment:
I think such case is supported by component itself, usually pagination is disabled when loading is in progress

Reply:
Good catch, I'll double check